src/darwin/Framework/CHIP/MTRDeviceController_Internal.h

```diff
@@ -14,26 +14,10 @@
  *    limitations under the License.
  */
 
-/**
- * Parts of MTRDeviceController that are not part of the framework API.  Mostly
- * for use from MTRDeviceControllerFactory.
- */
+#import <Matter/MTRDeviceController.h>
 
-#import <Foundation/Foundation.h>
 #import <Matter/MTRAccessGrant.h>
 #import <Matter/MTRBaseDevice.h> // for MTRClusterPath
-
-#import "MTRDeviceConnectionBridge.h" // For MTRInternalDeviceConnectionCallback
-#import "MTRDeviceController.h"
-
-#include <lib/core/CHIPError.h>
-#include <lib/core/DataModelTypes.h>
-
-#import "MTRBaseDevice.h"
-#import "MTRDeviceController.h"
-#import "MTRDeviceControllerDataStore.h"
-
-#import <Matter/MTRDefines.h>
```

Comment:
Why is it ok to not import MTRDefines here?  Because we know MTRDeviceController does?

I'm trying to understand the motivation for removing some of these imports of stuff the files actually use.

Reply:
I think it makes sense to view `Foo.h` and `Foo_Internal.h` as a sort of "stack", `Foo_Internal.h` always imports `Foo.h` at the start, and then adds more things to it, so I don't think there's any need to repeat imports from the public header in the internal one.

Reply:
Sounds good.